cli/tools/test/fmt.rs

```diff
@@ -74,3 +80,136 @@ pub fn format_test_error(js_error: &JsError) -> String {
+  };
+  (name.into(), action1, action2)
+}
+
+fn resource_close_hint(name: &str) -> &'static str {
+  match name {
+    "fsFile" => "Close the file handle by calling `file.close()`.",
+    "fetchRequest" => "Await the promise returned from `fetch()` or abort the fetch with an abort signal.",
+    "fetchRequestBody" => "Terminate the request body `ReadableStream` by closing or erroring it.",
+    "fetchResponse" => "Consume or close the response body `ReadableStream`, e.g `await resp.text()` or `await resp.body.cancel()`.",
+    "httpClient" => "Close the HTTP client by calling `httpClient.close()`.",
+    "dynamicLibrary" => "Unload the dynamic library by calling `dynamicLibrary.close()`.",
+    "httpConn" => "Close the inbound HTTP connection by calling `httpConn.close()`.",
+    "httpStream" => "Close the inbound HTTP request by responding with `e.respondWith()` or closing the HTTP connection.",
+    "tcpStream" => "Close the TCP connection by calling `tcpConn.close()`.",
+    "unixStream" => "Close the Unix socket connection by calling `unixConn.close()`.",
+    "tlsStream" => "Close the TLS connection by calling `tlsConn.close()`.",
+    "tlsListener" => "Close the TLS listener by calling `tlsListener.close()`.",
+    "unixListener" => "Close the Unix socket listener by calling `unixListener.close()`.",
+    "unixDatagram" => "Close the Unix datagram socket by calling `unixDatagram.close()`.",
+    "tcpListener" => "Close the TCP listener by calling `tcpListener.close()`.",
+    "udpSocket" => "Close the UDP socket by calling `udpSocket.close()`.",
+    "timer" => "Clear the timer by calling `clearInterval` or `clearTimeout`.",
+    "textDecoder" => "Close the text decoder by calling `textDecoder.decode('')` or `await textDecoderStream.readable.cancel()`.",
+    "messagePort" => "Close the message port by calling `messagePort.close()`.",
+    "webSocketStream" => "Close the WebSocket by calling `webSocket.close()`.",
+    "fsEvents" => "Close the file system watcher by calling `watcher.close()`.",
+    "childStdin" => "Close the child process stdin by calling `proc.stdin.close()`.",
+    "childStdout" => "Close the child process stdout by calling `proc.stdout.close()` or `await child.stdout.cancel()`.",
+    "childStderr" => "Close the child process stderr by calling `proc.stderr.close()` or `await child.stderr.cancel()`.",
+    "child" => "Close the child process by calling `proc.kill()` or `proc.close()`.",
+    "signal" => "Clear the signal listener by calling `Deno.removeSignalListener`.",
+    "stdin" => "Close the stdin pipe by calling `Deno.stdin.close()`.",
+    "stdout" => "Close the stdout pipe by calling `Deno.stdout.close()`.",
+    "stderr" => "Close the stderr pipe by calling `Deno.stderr.close()`.",
+    "compression" => "Close the compression stream by calling `await stream.writable.close()`.",
+    _ => "Close the resource before the end of the test.",
+  }
+}
```

Comment:
Consider doing these in a separate PR - in case we need to revert again, we'll keep diffs small and we know this code is ok

Reply:
Might even add relevant ops to reach for this data into Rust, so we can apply most of the changes from `40_test.js`?